### ISSUE TITLE
SPDX bom for twister builds

### ISSUE THREAD

**Issue by mkschreder** [zephyrproject-rtos/zephyr#70223]
**Is your feature request related to a problem? Please describe.**
I would like to get spdx bom generated for builds that are done with twister using sample.yaml file (while we are at it, why not add another accepted name for this file such as "product.yaml" and make that standard way to configure "variants"?)

**Describe the solution you'd like**
A flag for twister that would automatically initialize the spdx bom before the build and then generate the bom after the build of each defined variant. 

**Describe alternatives you've considered**
The only alternative today seems to be to use west build directly.

**Additional context**
None

**Comment by nordicjm** [comment]
> (while we are at it, why not add another accepted name for this file such as "product.yaml" and make that standard way to configure "variants"?)

No to this, sample.yaml is sufficient, we don't need more yaml files with the same thing

**Comment by kartben** [comment]
> > (while we are at it, why not add another accepted name for this file such as "product.yaml" and make that standard way to configure "variants"?)
> 
> No to this, sample.yaml is sufficient, we don't need more yaml files with the same thing

The way I understand the original comment is a suggestion to use a more representative name for this file, not to *add* a new file. And simply looking at e.g `example-application/app/sample.yaml` makes me agree that there might be an opportunity to improve the naming of this file to better reflect its actual use

**Comment by mkschreder** [comment]
Yes `example-application/app/sample.yaml` is a good example. sample.yaml makes it seem as though it is never intended to be used outside of samples directory. Another approach could be to merge sample.yaml and testcase.yaml into a single "twister.yaml". @nordicjm no to this as well from you?

**Comment by nordicjm** [comment]
They can be combined, the schema for both is the same: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/schemas/twister/testsuite-schema.yaml

**Comment by jeppenodgaard** [comment]
I am trying to fix this issue. Should I assign myself @nashif ?

**Comment by nashif** [comment]
> I am trying to fix this issue. Should I assign myself [@nashif](https://github.com/nashif) ?

fix what issue exactly?

**Comment by nashif** [comment]
> Yes `example-application/app/sample.yaml` is a good example. sample.yaml makes it seem as though it is never intended to be used outside of samples directory. Another approach could be to merge `sample.yaml` and `testcase.yaml` into a single `twister.yaml`. [@nordicjm](https://github.com/nordicjm) no to this as well from you?

you do not need to use sample.yaml above, you can also use testcase.yaml, those files define tests and are consumed by both twister and west, the naming might be unfortunate and could be better, but please do not call this twister.yaml, this is already being used elsewhere. I am fine just simply calling this tests.yaml or whatever, for both samples and tests and anything else.

**Comment by nashif** [comment]
see https://github.com/zephyrproject-rtos/zephyr/pull/107155

**Comment by jeppenodgaard** [comment]
> > I am trying to fix this issue. Should I assign myself [@nashif](https://github.com/nashif) ?
> 
> fix what issue exactly?

This issue (SPDX bom for twister builds).
